apis/v1alpha2/gateway_types.go

```diff
@@ -455,10 +492,45 @@ const (
 )
 
 const (
-	// This condition indicates whether the Listener has been
-	// configured on the Gateway.
+	// This condition indicates whether a Listener have generated some
+	// configuration that will soon be ready in the underlying data plane.
 	//
-	// Possible reasons for this condition to be true are:
+	// It is a positive-polarity summary condition, and so should always be
+	// present on the resource.
```

Comment:
Same comment as above, we should probably require ObservedGeneration here as well.